### ISSUE TITLE
Set properties boolean value to lowercase string

### ISSUE THREAD

**Issue by kevinjqliu** [apache/iceberg-python#919]
### Apache Iceberg version

None

## Issue
As shown in #895, setting a property to python's boolean value (`True`/`False`) will end up saving the boolean value as the string value `"True"`/`"False"`. This is because the property dictionary's values are converted to string by [`transform_dict_value_to_str` ](https://github.com/kevinjqliu/iceberg-python/blob/428b894aacb107547ba41433b4b6f37e1ad1914a/pyiceberg/types.py#L65-L70) implemented in #504.
So `True` -> `"True"`

## Fix
I can see this as a common footgun. Let's add a fix so that any boolean in the properties dictionary values are converted to its lowercase string. 

## Example
As a concrete example:
```
catalog.create_table(
    "default.example",
    schema=Schema(
        NestedField(1, "created_at", TimestampType(), required=True)
    ),
    properties={"write.parquet.bloom-filter-enabled.column.material_id": True},
)
```

Will set the property as 
```
{"write.parquet.bloom-filter-enabled.column.material_id": "True"}
```

But the expected value should be 
```
{"write.parquet.bloom-filter-enabled.column.material_id": "true"}
```

**Comment by soumya-ghosh** [comment]
@kevinjqliu could I pick this up?

**Comment by Fokko** [comment]
@soumya-ghosh assigned it to you 👍

**Comment by soumya-ghosh** [comment]
@Fokko @kevinjqliu PR https://github.com/apache/iceberg-python/pull/924 is ready for review.

**Comment by kevinjqliu** [comment]
Resolved in #924
